### PR TITLE
Add sub claim to the backend JWT.

### PR DESCRIPTION
**Pull request by ShalkiWenushika**
### Purpose
Add `sub` claim to the backend JWT

sample generated backend JWT contains following details:
```
{
  "sub": "admin",
  "http://wso2.org/claims/apiname": "Test",
  "http://wso2.org/claims/applicationtier": "Unlimited",
  "http://wso2.org/claims/version": "1.0.0",
  "http://wso2.org/claims/keytype": "PRODUCTION",
  "iss": "wso2.org/products/am",
  "http://wso2.org/claims/applicationname": "DefaultApplication",
  "http://wso2.org/claims/enduserTenantId": "0",
  "http://wso2.org/claims/applicationUUId": "b94f6c3d-f11b-4390-a9b9-6c7637aabaf7",
  "http://wso2.org/claims/subscriber": "admin",
  "azp": "4lIYZ4fU3tlUpJT26Q_kZ7RbkmAa",
  "http://wso2.org/claims/tier": "Unlimited",
  "scope": "default",
  "exp": 1632223,
  "http://wso2.org/claims/applicationid": "b94f6c3d-f11b-4390-a9b9-6c7637aabaf7",
  "http://wso2.org/claims/usertype": "Application_User",
  "iat": 1632224,
  "jti": "02ead499-707d-476a-8f9f-dd4c657e3aa4",
  "http://wso2.org/claims/apicontext": "/test/1.0.0"
}
```

### Issues

Related Issue https://github.com/wso2/product-microgateway/issues/2228

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [ ] Assigned 'Type' label
- [ ] Assigned the project
- [ ] Validated respective github issues
- [ ] Assigned milestone to the github issue(s)

### CODE REVIEW

File: enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java [hunk 1]
```diff
@@ -337,6 +337,11 @@ public static JWTInfoDto generateJWTInfoDto(JSONObject subscribedAPI, JWTValidat
         String apiVersion = requestContext.getMatchedAPI().getAPIConfig().getVersion();
         jwtInfoDto.setApiContext(apiContext);
         jwtInfoDto.setVersion(apiVersion);
+        if (jwtInfoDto.getJwtValidationInfo().getClaims() != null
+                && jwtInfoDto.getJwtValidationInfo().getClaims().get("sub") != null) {
+            String sub = (String) jwtInfoDto.getJwtValidationInfo().getClaims().get("sub");
+            jwtInfoDto.setSub(sub);
+        }
         constructJWTContent(subscribedAPI, apiKeyValidationInfoDTO, jwtInfoDto);
```
Comment (fazlan-nazeem) [inline enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java:345]: Don't we have the same value in apiKeyValidationInfoDTO's endusername?

Comment (ShalkiWenushika) [reply enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java:345]: Getting a null for apiKeyValidationInfoDTO's endusername. We are setting the endusername in here https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L140
But this dto is getting reset in this line: https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L148

After this line, endusername resets to null

Comment (ShalkiWenushika) [reply enforcer/src/main/java/org/wso2/choreo/connect/enforcer/util/FilterUtils.java:345]: Getting a null for apiKeyValidationInfoDTO's endusername. We are setting the endusername from this method: https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L140

But this dto get reset in this line: https://github.com/wso2/product-microgateway/blob/main/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java#L148

After this line, endusername resets to null.

**Comment by ShalkiWenushika** [comment]
Created a new PR in choreo branch: https://github.com/wso2/product-microgateway/pull/2230

**Pull request closed without merging**